src/overlays/actors/ovl_Bg_Mizu_Water/z_bg_mizu_water.c

```diff
@@ -21,19 +43,307 @@ const ActorInit Bg_Mizu_Water_InitVars = {
+                break;
+            case 2:
+                Flags_SetSwitch(globalCtx, 0x1E);
+                break;
+        }
+        bREG(0) = 0;
+    }
+    if ((Flags_GetSwitch(globalCtx, 0x1C)) && (switchFlag != 0x1C)) {
+        ret = 3;
+    } else if ((Flags_GetSwitch(globalCtx, 0x1D)) && (switchFlag != 0x1D)) {
+        ret = 2;
+    } else if ((Flags_GetSwitch(globalCtx, 0x1E)) && (switchFlag != 0x1E)) {
+        ret = 1;
+    } else {
+        ret = 0;
+    }
+
+    return ret;
+}
+
+void BgMizuWater_SetWaterBoxesHeight(WaterBox* waterBoxes, s16 height) {
+    u32 i;
+
+    for (i = 0; i < 8; i++) {
+        waterBoxes[waterBoxIDs[i]].unk_02 = height;
+    }
+}
+
+void BgMizuWater_Init(Actor* thisx, GlobalContext* globalCtx) {
+    BgMizuWater* this = THIS;
+    f32 initialActorY;
+    WaterBox* waterBoxes;
+    s32 waterLevelActionIndex;
+
+    waterBoxes = globalCtx->colCtx.stat.colHeader->waterBoxes;
+    this->type = this->actor.params & 0xFF;
+    this->switchFlag = (this->actor.params >> 8) & 0xFF;
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    initialActorY = this->actor.posRot.pos.y;
+    this->baseY = initialActorY;
+    this->targetY = initialActorY;
+
+    switch (this->type) {
+        case 0:
+            if (bREG(15) == 0) {
+                osSyncPrintf("<コンストラクト>%x %x %x\n", Flags_GetSwitch(globalCtx, 0x1C),
+                             Flags_GetSwitch(globalCtx, 0x1D), Flags_GetSwitch(globalCtx, 0x1E));
+            }
+            waterLevelActionIndex = BgMizuWater_GetWaterLevelActionIndex(-1, globalCtx);
+            this->actor.posRot.pos.y = waterLevels[waterLevelActionIndex].yDiff + this->baseY;
+            BgMizuWater_SetWaterBoxesHeight(waterBoxes, this->actor.posRot.pos.y);
+            this->actor.params = waterLevels[waterLevelActionIndex].switchFlag;
+            Flags_UnsetSwitch(globalCtx, 0x1C);
+            Flags_UnsetSwitch(globalCtx, 0x1D);
+            Flags_UnsetSwitch(globalCtx, 0x1E);
+
+            switch (this->actor.params) {
+                case 0x1E:
+                    Flags_SetSwitch(globalCtx, 0x1E);
+                    break;
+                case 0x1D:
+                    Flags_SetSwitch(globalCtx, 0x1D);
+                    break;
+                case 0x1C:
+                default:
+                    Flags_SetSwitch(globalCtx, 0x1C);
+                    break;
+            }
+            this->targetY = this->actor.posRot.pos.y;
+            break;
+        case 1:
+            break;
+        case 2:
+            if (Flags_GetSwitch(globalCtx, this->switchFlag)) {
+                this->actor.posRot.pos.y = this->baseY + 85.0f;
+            }
+            waterBoxes[6].unk_02 = this->actor.posRot.pos.y;
+            break;
+        case 3:
+            if (Flags_GetSwitch(globalCtx, this->switchFlag)) {
+                this->actor.posRot.pos.y = this->baseY + 110.0f;
+                if (1) {}
+                this->targetY = this->actor.posRot.pos.y;
```

Comment:
Any chance this works? (might still need the if (1) {} for other reasons). If it works you can apply the same idea in case 4 below.
```suggestion
                this->targetY = this->actor.posRot.pos.y = this->baseY + 110.0f;
```

Reply:
It doesn't work. It's strange, but after the calculation of `this->baseY + 110.0f` the result is stored in `this->actor.posRot.pos.y` and then afterwards loaded again from there to store it in `this->targetY`. That's why the  `if (1) {}` is required.